clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1043,11 +997,7 @@ void KernelObjVisitor::VisitRecord(const CXXRecordDecl *Owner, ParentTy &Parent,
 
 // A base type that the SYCL OpenCL Kernel construction task uses to implement
 // individual tasks.
-class SyclKernelFieldHandler {
-protected:
-  Sema &SemaRef;
-  SyclKernelFieldHandler(Sema &S) : SemaRef(S) {}
-
+class SyclKernelFieldHandlerBase {
```

Comment:
Essentially, I broke up SyclKernelFieldHandler into 2, with most of it being 'base'.  The 'base' is exactly the same as SyclKernelFieldHandler, except it no longer has a reference to an instance of Sema.  This allows us to create the no-op handler.